First we need to load the dataset 

In [13]:
import pandas as pd 


In [14]:
df = pd.read_csv("Evaluation-dataset.csv")
print(df.shape)
df.head(10)

(10131, 15)


Tires where delivered to the garage of my choice,the garage notified me when they had been delivered. A day and time was arranged with the garage and I went and had them fitted,a Hassel free experience.  \
0  Easy Tyre Selection Process, Competitive Prici...                                                                                                                                                           
1         Very easy to use and good value for money.                                                                                                                                                           
2              Really easy and convenient to arrange                                                                                                                                                           
3  It was so easy to select tyre sizes and arrang...                                                                                                                                                           
4  service was excellent. Only slight downside wa...                                                                                                                                                           
5  User friendly Website. Competitive Prices. Goo...                                                                                                                                                           
6                       Excellent prices and service                                                                                                                                                           
7  It was very straightforward and the garage was...                                                                                                                                                           
8                               Use of local garage.                                                                                                                                                           
9                      Easy to use, also good price.                                                                                                                                                           

      garage service positive  ease of booking positive  \
0     garage service positive  value for money positive   
1    value for money positive                       NaN   
2    ease of booking positive                       NaN   
3           location positive  value for money positive   
4  length of fitting positive  ease of booking positive   
5     garage service positive  value for money positive   
6    value for money positive                       NaN   
7     garage service positive                       NaN   
8           location positive                       NaN   
9    value for money positive                       NaN   

                 Unnamed: 3 Unnamed: 4 Unnamed: 5 Unnamed: 6 Unnamed: 7  \
0                       NaN        NaN        NaN        NaN        NaN   
1                       NaN        NaN        NaN        NaN        NaN   
2                       NaN        NaN        NaN        NaN        NaN   
3  ease of booking positive        NaN        NaN        NaN        NaN   
4  ease of booking negative        NaN        NaN        NaN        NaN   
5                       NaN        NaN        NaN        NaN        NaN   
6                       NaN        NaN        NaN        NaN        NaN   
7                       NaN        NaN        NaN        NaN        NaN   
8                       NaN        NaN        NaN        NaN        NaN   
9                       NaN        NaN        NaN        NaN        NaN   

  Unnamed: 8 Unnamed: 9 Unnamed: 10 Unnamed: 11 Unnamed: 12 Unnamed: 13  \
0        NaN        NaN         NaN         NaN         NaN         NaN   
1        NaN        NaN         NaN         NaN         NaN         NaN   
2        NaN

outputting the dataset , we can make find some things : 
1. There is statement in first column and some comment type us there in front if the statement .
2. I can change the first columns name since its too big .
3 this is the shape (10131, 15) . 15 columns and 10131 rows .


In [15]:
#changing the columns name  : 
df = df.rename(columns= {"Tires where delivered to the garage of my choice,the garage notified me when they had been delivered. A day and time was arranged with the garage and I went and had them fitted,a Hassel free experience." : " Statement 1 "})

In [16]:
df

Statement 1   \
0      Easy Tyre Selection Process, Competitive Prici...   
1             Very easy to use and good value for money.   
2                  Really easy and convenient to arrange   
3      It was so easy to select tyre sizes and arrang...   
4      service was excellent. Only slight downside wa...   
...                                                  ...   
10126  I ordered the wrong tyres, however [REDACTED] ...   
10127  Good experience, first time I have used [REDAC...   
10128  I ordered the tyre I needed on line, booked a ...   
10129  Excellent service from point of order to fitti...   
10130  Seamless, well managed at both ends. I would r...   

          garage service positive       ease of booking positive  \
0         garage service positive       value for money positive   
1        value for money positive                            NaN   
2        ease of booking positive                            NaN   
3               location positive       value for money positive   
4      length of fitting positive       ease of booking positive   
...                           ...                            ...   
10126             refund positive  delivery punctuality positive   
10127  length of fitting positive                            NaN   
10128           location positive  delivery punctuality positive   
10129                         NaN                            NaN   
10130                         NaN                            NaN   

                       Unnamed: 3                Unnamed: 4 Unnamed: 5  \
0                             NaN                       NaN        NaN   
1                             NaN                       NaN        NaN   
2                             NaN                       NaN        NaN   
3        ease of booking positive                       NaN        NaN   
4        ease of booking negative                       NaN        NaN   
...                           ...                       ...        ...   
10126   refund timescale positive                       NaN        NaN   
10127                         NaN                       NaN        NaN   
10128  length of fitting positive  value for money positive        NaN   
10129                         NaN                       NaN        NaN   
10130                         NaN                       NaN        NaN   

      Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9 Unnamed: 10 Unnamed: 11  \
0            NaN        NaN        NaN        NaN         NaN         NaN   
1            NaN        NaN        NaN        NaN         NaN         NaN   
2            NaN        NaN        NaN        NaN         NaN         NaN   
3            NaN        NaN        NaN        NaN         NaN         NaN   
4            NaN        NaN        NaN        NaN         NaN         NaN   
...          ...        ...        ...        ...         ...         ...   
10126        NaN        NaN        NaN        NaN         NaN         NaN   
10127        NaN        NaN        NaN        NaN         NaN         NaN   
10128        NaN        NaN        NaN        NaN         NaN         NaN   
10129        NaN        NaN        NaN        NaN         NaN         NaN   
10130        NaN        NaN        NaN        NaN         NaN         NaN   

      Unnamed: 12 Unnamed: 13 Unnamed: 14  
0             NaN         NaN         NaN  
1             NaN         NaN         NaN  
2             NaN         NaN         NaN  
3             NaN         NaN         NaN  
4             NaN         NaN         NaN  
...           ...         ...         ...  
10126         NaN         NaN         NaN  
10127         NaN         NaN         NaN  
10128         NaN         NaN         NaN  
10129         NaN         NaN         NaN  
10130         NaN         NaN         NaN  

[10131 rows x 15 columns]

The context in the Dataset are repaeating , we can count their frequency . We will need a datastructure for it . A map wil do the work 